src/pages/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../../components/Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { PeopleTable } from '../../components/PeopleTable';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { LoadingError } from '../../components/LoadingError';
+import { PeopleFilters } from '../../components/PeopleFilters';
+import { normalize } from '../../utils/stringManipulation';
+import { Orders, SortKeys } from '../../types/sortTypes';
+import { NoPeopleMessage } from '../../components/NoPeopleMessage';
+import { NoVisiblePeople } from '../../components/NoVisiblePeople';
+
+type FilterParams = {
+  query: string;
+  sex: string;
+  centuries: string[];
+};
+
+const getFilteredPeople = (people: Person[], params: FilterParams) => {
+  const { query, sex, centuries } = params;
+
+  let peopleCopy = [...people];
+
+  if (query) {
+    peopleCopy = peopleCopy.filter(person => {
+      const normalizedQuery = normalize(query);
+
+      return (
+        normalize(person.name).includes(normalizedQuery) ||
+        normalize(person.fatherName ?? '-').includes(normalizedQuery) ||
+        normalize(person.motherName ?? '-').includes(normalizedQuery)
+      );
+    });
+  }
+
+  if (sex) {
+    peopleCopy = peopleCopy.filter(person => person.sex === sex);
+  }
+
+  if (centuries.length !== 0) {
+    peopleCopy = peopleCopy.filter(person => {
+      const centurie = Math.ceil(person.born / 100);
+
+      return centuries.includes(String(centurie));
+    });
+  }
+
+  return peopleCopy;
+};
+
+type SortParams = {
+  order: string;
+  sort: string;
+};
+
+const getSortedPeople = (people: Person[], params: SortParams) => {
```

Comment:
it is better to take out in a separate file

Reply:
good point